src/engine/SCons/Tool/msvs.py

```diff
@@ -513,11 +538,13 @@ def __init__(self, dspfile, source, env):
         for n in sourcenames:
             self.sources[n].sort(key=lambda a: a.lower())
 
-        def AddConfig(self, variant, buildtarget, outdir, runfile, cmdargs, dspfile=dspfile):
+        def AddConfig(self, variant, buildtarget, outdir, runfile, cmdargs, cppdefines=[], cpppaths=[], dspfile=dspfile):
```

Comment:
Isn't having default arg values be a list considered bad practice? (Because lists are mutable and will persist between function calls?)

Reply:
Good point, hadn't realized this. Will fix.

Reply:
I think the recommend is arg=None, and then in logic, if arg is None, arg =[]
That way changes to defaults don't persist.

Reply:
grin... my teachings taking hold?